src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -222,42 +253,47 @@ library BeaconChainProofs {
             "BeaconChainProofs.verifyWithdrawalProofs: executionPayloadProof has incorrect length");
         require(proofs.slotProof.length == 32 * (BEACON_BLOCK_HEADER_FIELD_TREE_HEIGHT),
             "BeaconChainProofs.verifyWithdrawalProofs: slotProof has incorrect length");
-        require(proofs.blockNumberProof.length == 32 * (EXECUTION_PAYLOAD_HEADER_FIELD_TREE_HEIGHT),
-            "BeaconChainProofs.verifyWithdrawalProofs: blockNumberProof has incorrect length");
-
-
-        /**
-         * Computes the block_header_index relative to the beaconStateRoot.  It concatenates the indexes of all the
-         * intermediate root indexes from the bottom of the sub trees (the block header container) to the top of the tree
-         */
-        uint256 blockHeaderIndex = BLOCK_ROOTS_INDEX << (BLOCK_ROOTS_TREE_HEIGHT)  | uint256(proofs.blockHeaderRootIndex);
-        // Verify the blockHeaderRoot against the beaconStateRoot
-        require(Merkle.verifyInclusionSha256(proofs.blockHeaderProof, beaconStateRoot, proofs.blockHeaderRoot, blockHeaderIndex),
-            "BeaconChainProofs.verifyWithdrawalProofs: Invalid block header merkle proof");
+        require(proofs.timestampProof.length == 32 * (EXECUTION_PAYLOAD_HEADER_FIELD_TREE_HEIGHT),
+            "BeaconChainProofs.verifyWithdrawalProofs: timestampProof has incorrect length");
+
+        {
+            /**
+            * Computes the block_header_index relative to the beaconStateRoot.  It concatenates the indexes of all the
+            * intermediate root indexes from the bottom of the sub trees (the block header container) to the top of the tree
+            */
+            uint256 blockHeaderIndex = BLOCK_ROOTS_INDEX << (BLOCK_ROOTS_TREE_HEIGHT)  | uint256(proofs.blockHeaderRootIndex);
+            // Verify the blockHeaderRoot against the beaconStateRoot
+            require(Merkle.verifyInclusionSha256(proofs.blockHeaderProof, beaconStateRoot, proofs.blockHeaderRoot, blockHeaderIndex),
+                "BeaconChainProofs.verifyWithdrawalProofs: Invalid block header merkle proof");
```

Comment:
is this is same as the logic in ` verifyStateRootAgainstLatestBlockHeaderRoot` ?
seems like it could be missing the proof length check?

Reply:
No its a different proof, this is the check we need:

`require(proof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT), "BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot: Proof has incorrect length");`

Reply:
> No its a different proof, this is the check we need:
> 
> `require(proof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT), "BeaconChainProofs.verifyStateRootAgainstLatestBlockHeaderRoot: Proof has incorrect length");`

sorry, am still confused. Isn't this literally exactly the check that I said was missing?

Reply:
Oh  you're saying we need it here - I thought you meant we need it in the other spot.  Either way we don't need that proof length check here because they are different proofs all together

Reply:
> Oh you're saying we need it here - I thought you meant we need it in the other spot. Either way we don't need that proof length check here because they are different proofs all together

I'm still wondering what you thought I meant.
Looks like the distinction between these proofs is that this one -- inside `verifyWithdrawalProofs` -- is a proof for a *general* block header, whereas inside of `verifyStateRootAgainstLatestBlockHeaderRoot`, it is specifically a proof of the `latest block header root` field.

And the relevant check that you are performing for the highlighted proof (in `verifyWithdrawalProofs`) is
```
        require(proofs.blockHeaderProof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT + BLOCK_ROOTS_TREE_HEIGHT),
            "BeaconChainProofs.verifyWithdrawalProofs: blockHeaderProof has incorrect length");
 ```
right? (if this isn't right, then I do indeed think there's an issue here. this isn't rhetorical, please tell me if this is right or not.)
 
I realize this is subtle but IMO this needing such a long discussion suggests to me that this could be _way_ clearer in the documentation.

Reply:
> And the relevant check that you are performing for the highlighted proof (in `verifyWithdrawalProofs`) is
> 
> ```
>         require(proofs.blockHeaderProof.length == 32 * (BEACON_STATE_FIELD_TREE_HEIGHT + BLOCK_ROOTS_TREE_HEIGHT),
>             "BeaconChainProofs.verifyWithdrawalProofs: blockHeaderProof has incorrect length");
> ```
> 
> right? (if this isn't right, then I do indeed think there's an issue here. this isn't rhetorical, please tell me if this is right or not.)
yes exactly.  Again probably a confusion brought on by the similar naming.